core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -201,7 +215,9 @@ public static RecordState forId(byte id) {
         this.lock = new ReentrantReadWriteLock();
         // Should be initialized when the partition is loaded by reading state from the share persister.
         this.nextFetchOffset = 0;
-        this.endOffset = 0;
+        // If startOffset and endOffset are both -1, this means cachedState is empty
+        this.startOffset = -1;
+        this.endOffset = -1;
```

Comment:
Nit: Cached state can be looked individually to see if it's empty. Shall these not be 0, though either works.